### ISSUE TITLE
Polygon Redaction Tool

### ISSUE THREAD

**Issue by liseandtea** [bcgov/foi-flow#3654]
* As an analyst
* I want to be able to draw a custom-shaped redaction box
* so that I can redact information not able to be captured within the rectangular-shaped redaction tool. 

**Assumptions & Scope**	
What are the assumptions for this story?

What is IN scope?

What is NOT in scope?

**Acceptance Criteria**	

**Scenario 1: Drawing a Custom-Shaped Redaction Box.**
* GIVEN An analyst is on a page where the information needed to be redacted cannot be captured by using the rectangular-shaped redaction tool 
* WHEN the analyst selects the polygon redaction tool
* AND begins to custom-create a shape
* THEN they are able to draw around the information needed to be redacted
*AND all of the information contained within the custom shape is withheld.

**This is blocked by #3524 ** 

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well-articulated User Story?
2. [ ] Is there Acceptance Criteria that cover all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, that passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by liseandtea** [comment]
@lmullane @m-prodan @KyEggleston let me know if I missed anything!

**Comment by KyEggleston** [comment]
@liseandtea @m-prodan @lmullane - Just to confirm, we'd want the polygon redaction tool functionality to mimic the polygon 'shapes' tool in the redaction app? This would probably be the easiest, if feasible. Can add this to the assumptions/scope of this ticket. 

![image](https://user-images.githubusercontent.com/117312323/232150098-01830703-2ca0-46b7-b8a2-732cdc9f688c.png)

**Comment by liseandtea** [comment]
@KyEggleston @lmullane @m-prodan Yes, we want to mimic the same capability as the polygon drawing tool in "Shapes"

**Comment by lmullane** [comment]
@KyEggleston, to confirm AC matches how the polygon tool works in shapes.


 @liseandtea to update AC to match the what happens when you redact with the FOIPPA modal

@m-prodan

**Comment by liseandtea** [comment]
@lmullane @KyEggleston @m-prodan updated the AC for this ticket. Let me know if it makes sense or if anything is missing.

**Comment by m-prodan** [comment]
Hi Lise - nice work on the updates. I made a slight tweak to scenario 5, but this looks pretty good to me!

**Comment by lmullane** [comment]
Thanks for all of your work on this, @liseandtea. I have made some edits to the AC and assumptions.

**Comment by lmullane** [comment]
Need to complete the research task #3524 before an estimate can be made. As likely we will need to do custom development.

**Comment by lmullane** [comment]
Need to complete the research task #3524 before an estimate can be made. As likely we will need to do custom development.

**Comment by abin-aot** [comment]
Refinement comments
- This is a research topic using PDFTron API and/or using a Back End redaction with Shape coordinates - @abin-aot research tells from #3524

**Comment by KyEggleston** [comment]
Decision is to wait to see if this feature will be developed by vendor, rather than custom development at the moment.
